pandas/core/reshape/merge.py

```diff
@@ -492,6 +492,10 @@ def __init__(self, left, right, how='inner', on=None,
                  left_index=False, right_index=False, sort=True,
                  suffixes=('_x', '_y'), copy=True, indicator=False,
                  validate=None):
+        if isinstance(left, Series):
+            left = left.to_frame()
+        if isinstance(right, Series):
+            right = right.to_frame()
```

Comment:
need to validate that the input is a Series or Dataframe and nothing else

Reply:
Thanks - Had initially addeded these validations at 525 and also before 541 (if memory serves right) on my local machine - but then it fails at 571...

Thus after trying to fix the same issue (checking if input is Series or DataFrame) at multiple places, the current change seemed like the most logical fix as we are changing the type of incoming Series object to DataFrame and then avoiding any susequent changes to the code....

However open to suggestions...

Reply:
this is ok, but what I mean is validation that you have a Series/DataFrame and nothing else. (and a test to go with that), something like

```
left = validate_operand(left)
right = validate_operand(right)


......

def validate_operand(obj):
     if isinstance(obj, DataFrame): 
         return obj
     elif isinstance(obj, Series):
        return obj.to_frame()
     raise ValueError(.....)
```

Reply:
also need to test this with left_index=True, right_index=True (which should be fine). And a test that a Series w/o a name raises (it will convert just fine, but then you would have to specify the on in a funny way)

Reply:
Thanks for the review - done.
also raises a ValueError for Series w/o a name